rpcs3/rpcs3qt/gui_settings.cpp

```diff
@@ -117,23 +117,22 @@ void gui_settings::ShowBox(QMessageBox::Icon icon, const QString& title, const Q
 
 	const QFlags<QMessageBox::StandardButton> buttons = icon != QMessageBox::Information ? QMessageBox::Yes | QMessageBox::No : QMessageBox::Ok;
 
-	QMessageBox* mb = new QMessageBox(icon, title, text, buttons, parent, Qt::Dialog | Qt::MSWindowsFixedSizeDialogHint | (always_on_top ? Qt::WindowStaysOnTopHint : Qt::Widget));
-	mb->deleteLater();
-	mb->setTextFormat(Qt::RichText);
+	QMessageBox mb(icon, title, text, buttons, parent, Qt::Dialog | Qt::MSWindowsFixedSizeDialogHint | (always_on_top ? Qt::WindowStaysOnTopHint : Qt::Widget));
+	mb.setTextFormat(Qt::RichText);
 
 	if (has_gui_setting && icon != QMessageBox::Critical)
 	{
-		mb->setCheckBox(new QCheckBox(tr("Don't show again")));
+		mb.setCheckBox(new QCheckBox(tr("Don't show again"), &mb));
```

Comment:
setting the parent here is a placebo. setCheckBox already explicitly passes the ownership to mb